api/meetings/views.py

```diff
@@ -0,0 +1,240 @@
+    required_write_scopes = [CoreScopes.NULL]
+    model = Conference
+
+    # This view goes under the _/ namespace
+    versioning_class = PrivateVersioning
+
+    serializer_class = MeetingSerializer
+    view_category = 'meetings'
+
+
+class MeetingList(BaseMeetingView, generics.ListAPIView, ListFilterMixin):
+
+    view_name = 'meeting-list'
+    ordering = ('-modified', )  # default ordering
+
+    # overrides ListFilterMixin
+    def get_default_queryset(self):
+        conferences = Conference.objects.filter(is_meeting=True)
+        for conference in conferences:
+            # caching num_submissions on the Conference objects
+            conference.num_submissions = conference.submissions.count()
+            conference.save()
+        return conferences.filter(num_submissions__gte=settings.CONFERENCE_MIN_COUNT)
+
+    # overrides ListAPIView
+    def get_queryset(self):
+        return self.get_queryset_from_request()
+
+
+class MeetingDetail(BaseMeetingView, generics.RetrieveAPIView):
+
+    view_name = 'meeting-detail'
+
+    def get_object(self):
+        # No minimum submissions count for accessing meeting directly
+        return self.get_meeting()
+
+
+class BaseMeetingSubmission(JSONAPIBaseView, MeetingMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        IsPublic,
+    )
+
+    required_read_scopes = [CoreScopes.MEETINGS_READ, CoreScopes.NODE_BASE_READ]
+    required_write_scopes = [CoreScopes.NULL]
+    model = AbstractNode
+
+    # This view goes under the _/ namespace
+    versioning_class = PrivateVersioning
+
+    serializer_class = MeetingSubmissionSerializer
+    view_category = 'meetings'
+
+    def get_serializer_context(self):
+        context = super(BaseMeetingSubmission, self).get_serializer_context()
+        context['meeting'] = self.get_meeting()
+        return context
+
+
+class MeetingSubmissionList(BaseMeetingSubmission, generics.ListAPIView, ListFilterMixin):
+    view_name = 'meeting-submissions'
+
+    ordering = ('-created', )  # default ordering
+    ordering_fields = ('title', 'meeting_category', 'author_name', 'download_count', 'created', )
+
+    # overrides ListFilterMixin
+    def get_default_queryset(self):
+        # Returning public meeting submissions that have at least one file attached
+        meeting = self.get_meeting()
+        queryset = meeting.submissions.filter(
+            files__type='osf.osfstoragefile',
+            files__deleted_on__isnull=True,
+        ).annotate(
+            annotated_file_count=Count('files'),
+        ).filter(annotated_file_count__gte=1)
+        return self.annotate_queryset_for_filtering_and_sorting(meeting, queryset)
+
+    # overrides ListAPIView
+    def get_queryset(self):
+        return self.get_queryset_from_request()
+
+    def build_query_from_field(self, field_name, operation):
+        if field_name == 'author_name':
```

Comment:
yes, b/c it's a SerializerMethodField.  You can only do `.annotate()` on a queryset, but I guess I could temporarily set it as a string on the python object on the DetailView:
```python

    def get_object(self):
        meeting = self.get_meeting()
        node = self.get_node()
        # Submission must be associated with the Conference
        if meeting.endpoint not in node.tags.values_list('name', flat=True):
            raise NotFound('This is not a submission to {}.'.format(meeting.name))
        node.author_name = <move author_name calculation here>
        return node
```

Reply:
is this what you're talking about?

Reply:
Yeah, something like that.

Reply:
I tried this, it pulls most of the SerializerMethodField logic to the DetailView since there still needs to be some calculating to build author_id, author_name, and meeting_category.  So as far as lines reduced, it really only gets rid of `build_query_from_field` - my opinion is that since overriding `build_query_from_field` to handle filtering SerializerMethodFields is a common pattern, I'd rather stick with existing app conventions - but I could be convinced -